File: 채주혁/3주차/programmers_올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Stack<String> stack = new Stack<>();
        for(int i=0; i<s.length(); i++) {
            String str = s.substring(i,i+1);
            if(str.equals("(")) {
                stack.push(str);
            }
            else {
                if(!stack.isEmpty() && stack.peek().equals("(")) {
                    stack.pop();
                }
                else {
                    answer=false;
                    return answer;
                }
            }
        }
        if(!stack.isEmpty()) {
            answer=false;
        }

        return answer;
    }
}

File: 신혜원/week3/programmers_ 올바른괄호.js
function solution(s){
    let stack = [];

    for (let i = 0; i < s.length; i++) {
        if (s[i] === "(") { // 여는 괄호면 스택에 추가
            stack.push("(");
        } else { // 닫는 괄호면
            if (stack.length === 0) { // 스택이 비어있다면 false 반환
                return false;
            }
            stack.pop(); // 스택에서 () 제거
        }
    }

    return stack.length === 0; // 스택이 비어있으면 true, 그렇지 않으면 false 반환
}